src/pages/FeeRateTracker/index.tsx

```diff
@@ -0,0 +1,351 @@
+            .div(feeRates.length)
+            .dp(1, BigNumber.ROUND_HALF_EVEN)
+            .toNumber()
+        : 0
+    })
+    return { confirmationTimeFeeRateX, confirmationTimeFeeRateY }
+  }, [transactionFees.transactionFeeRates])
+
+  const { minFeeRate, maxFeeRate } = useMemo(() => {
+    let minFeeRate = Number.MAX_SAFE_INTEGER
+    let maxFeeRate = Number.MIN_SAFE_INTEGER
+    transactionFees.pendingTransactionFeeRates.forEach(r => {
+      if (r.feeRate < minFeeRate) minFeeRate = r.feeRate
+      if (r.feeRate > maxFeeRate) maxFeeRate = r.feeRate
+    })
+    return { minFeeRate, maxFeeRate }
+  }, [transactionFees.pendingTransactionFeeRates])
+
+  const rootFeeRateTransactionCountX = useMemo(() => {
+    if (maxFeeRate < minFeeRate) {
+      return []
+    }
+    return [...Array(Math.floor(maxFeeRate * 10) - Math.floor(minFeeRate * 10) + 1).keys()].map(
+      i => (i + Math.floor(minFeeRate * 10)) / 10,
+    )
+  }, [minFeeRate, maxFeeRate])
+
+  const feeRateTransactionCountX = useMemo(() => {
+    if (feeRatePrecision === 1 || !feeRateCeilFilter) {
+      return rootFeeRateTransactionCountX
+    }
+    const unit = new BigNumber(10).pow(-feeRatePrecision)
+    return [...Array(10).keys()].map(i => feeRateCeilFilter.plus(unit.multipliedBy(i)).toString())
+  }, [rootFeeRateTransactionCountX, feeRatePrecision, feeRateCeilFilter])
+
+  const feeRateTransactionCountY = useMemo(
+    () =>
+      feeRateTransactionCountX.map(
+        tc =>
+          transactionFees.pendingTransactionFeeRates.filter(fr =>
+            new BigNumber(fr.feeRate)
+              .dp(feeRatePrecision, BigNumber.ROUND_FLOOR)
+              .isEqualTo(new BigNumber(tc).dp(feeRatePrecision, BigNumber.ROUND_FLOOR)),
+          ).length,
+      ),
+    [transactionFees.pendingTransactionFeeRates, feeRatePrecision, feeRateTransactionCountX],
+  )
+
+  useEffect(() => {
+    if (feeRatePrecision !== 1 && !feeRateTransactionCountY.find(y => y !== 0)) {
+      setFeeRatePrecision(1)
+      setFeeRateCeilFilter(null)
+    }
+  }, [feeRatePrecision, feeRateTransactionCountY])
+
+  return (
+    <Content>
+      <div className={`${styles.feeRateTrackerPanel} container`}>
+        <Row className={styles.title}>Fee Rate Tracker</Row>
+        <Row className={styles.updatedTimeCounter}>Updated {fetchedTimeCounter}s ago</Row>
+        <Row justify="space-evenly" className={styles.priceCenter}>
+          {feeRateCards.map(feeRateCard => (
+            <Col span={6} key={feeRateCard.limitTimeSeconds}>
+              <div className={styles.card}>
+                <div>{feeRateCard.priority}</div>
+                <div>{feeRateCard.feeRate} shannons/byte</div>
+                <div>{feeRateCard.limitTimeDisplay}</div>
+              </div>
+            </Col>
+          ))}
+        </Row>
+        <Row className={styles.chartTitle}>Confirmation Time x Fee Rate (Last 10000 transactions)</Row>
+        <Row className={styles.chart}>
+          <ReactEchartsCore
```

Comment:
I suggest that these 3 echarts be extracted into 3 separate components and passed in TransactionFeesStatistic and other necessary data to be used, so as to optimize readability.

Reply:
> I suggest that these 3 echarts be extracted into 3 separate components and passed in TransactionFeesStatistic and other necessary data to be used, so as to optimize readability.

Well, that's what I want too